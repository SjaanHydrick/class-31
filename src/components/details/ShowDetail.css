@import url('https://fonts.googleapis.com/css2?family=Nunito:wght@400;900&display=swap');

@import url('https://fonts.googleapis.com/css2?family=Abril+Fatface&display=swap');

.dark {
    background-color: darkcyan;
    border: solid 1px darkcyan;
}

.showDetail {
    text-align: center;
    margin: 10%;
    margin-top: 5%;
}

.detailsImg {
    height: 500px;
}

.detailsName {
    font-family: 'Abril Fatface';
    font-size: 40pt;
}

.dark .detailsName, .dark .description {
    color: cornsilk;
}

.status {
    font-family: 'Nunito';
    font-weight: 900;
    font-size: 15pt;
}

.dark .status {
    color: salmon;
}

.description {
    font-family: 'Nunito';
    font-size: 14pt;
}