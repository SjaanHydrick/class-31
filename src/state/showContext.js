import React, { createContext, useState, useContext } from 'react';

export const ThemeContext = createContext(null);

// eslint-disable-next-line react/prop-types
export const ThemeProvider = ({ children }) => {
  const [theme, setTheme] = useState('light');

  const toggle = ({ target }) => {
    if(target.checked) setTheme(() => 'dark');
    else setTheme(() => 'light');
  };

  return (
    <ThemeContext.Provider value={{ theme, setTheme, toggle }}>
      {children}
    </ThemeContext.Provider>
  );
};

export const useTheme = () => {
  const { theme } = useContext(ThemeContext);
  const { setTheme } = useContext(ThemeContext);
  const { toggle } = useContext(ThemeContext);
  return { theme, setTheme, toggle };
};
