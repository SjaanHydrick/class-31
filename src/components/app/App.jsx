import React, { useState } from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch
} from 'react-router-dom';
import Header from '../header/Header';
import ShowById from '../../containers/ShowById';
import ShowsPage from '../../containers/ShowsPage';
import { ThemeProvider } from '../../state/showContext';

export default function App() {
  const [Provider] = useState(() => ThemeProvider);  

  return (
    <div>
      <Router>
        <Provider>
          <Header />
          <Switch>
            <Route exact path="/" component={ShowsPage} />
            <Route path="/shows/:id" component={ShowById} />
          </Switch>
        </Provider>
      </Router>
    </div>
  );
}
