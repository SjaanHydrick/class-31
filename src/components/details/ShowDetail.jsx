import React from 'react';
import PropTypes from 'prop-types';
import { useTheme } from '../../state/showContext';
import styles from './ShowDetail.css';

const ShowDetail = ({ name, status, description, image }) => {
  const { theme } = useTheme();

  return (
    <div className={styles[theme]}>
      <div className={styles.showDetail}>
        <img src={image} alt={name} className={styles.detailsImg}/>
        <h2 className={styles.detailsName}>{name}</h2>
        <p className={styles.status}>Status: {status}</p>
        <p className={styles.description}>{description}</p>
      </div>
    </div>
  );
};

ShowDetail.propTypes = {
  image: PropTypes.string,
  name: PropTypes.string,
  status: PropTypes.string,
  description: PropTypes.string
};

export default ShowDetail;
