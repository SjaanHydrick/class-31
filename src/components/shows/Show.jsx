import React from 'react';
import PropTypes from 'prop-types';
import { useTheme } from '../../state/showContext';
import styles from './Shows.css';

const Show = ({ image, name }) => {
  const { theme } = useTheme();

  return (
    <div className={`${styles[theme]}`}>
      <div className={styles.show}>
        <img src={image} alt={name} className={styles.image}/>
        <h2 className={styles.name}>{name}</h2>
      </div>
    </div>
  );
};



Show.propTypes = {
  image: PropTypes.string,
  name: PropTypes.string
};

export default Show;
