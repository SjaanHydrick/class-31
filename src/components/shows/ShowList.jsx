import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Show from './Show';
import styles from './Shows.css';

const ShowList = ({ shows }) => {
  const showElements = shows.map(show => (
    <div key={show.id}>
      <Link to={`/shows/${show.id}`}>
        <Show image={show.image} name={show.name} />
      </Link>
    </div>
  ));

  return (
    <div className={styles.showList} data-testid="shows">
      {showElements}
    </div>
  );
};

ShowList.propTypes = {
  shows: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    image: PropTypes.string
  }))
};

export default ShowList;
