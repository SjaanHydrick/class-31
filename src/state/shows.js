import { useState, useEffect } from 'react';
import { fetchShows, fetchShowById } from '../services/shows';

export const useShows = () => {
  const [loading, setLoading] = useState(true);
  const [shows, setShows] = useState([]);
  const [count, setCount] = useState(1);

  useEffect(() => {
    fetchShows(count)
      .then(shows => {
        setShows(shows);
        setLoading(false);
      });
  }, [count]);

  return {
    loading,
    shows,
    count,
    setCount
  };
};

export const useShowById = id => {
  const [loading, setLoading] = useState(true);
  const [show, setShow] = useState(null);

  useEffect(() => {
    fetchShowById(id)
      .then(show => {
        setShow(show);
        setLoading(false);
      });
  }, [id]);

  return {
    loading,
    show
  };
};
