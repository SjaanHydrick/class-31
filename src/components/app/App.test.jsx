/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import App from './App';
import Show from '../shows/Show.jsx';
import ShowList from '../shows/ShowList.jsx';
import ShowDetail from '../details/ShowDetail.jsx';
import { ThemeProvider } from '../../state/showContext';

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders Show', () => {
    
    const arrow = [{
      'id': 29560,
      'name': 'Arrow',
      'image': 'https://static.episodate.com/images/tv-show/thumbnail/29560.jpg',
    }];

    const { asFragment } = render(
      <ThemeProvider>
        <BrowserRouter>
          <Show {...arrow} />
        </BrowserRouter>
      </ThemeProvider>
    );

    expect(asFragment()).toMatchSnapshot();

  });

  it('renders Shows', () => {
    const shows = [{ id: 1, name: '', image: '' }];
    const { asFragment } = render(
      <ThemeProvider>
        <BrowserRouter>
          <ShowList shows={shows}/>
        </BrowserRouter>
      </ThemeProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders ShowDetail', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <BrowserRouter>
          <ShowDetail path={'shows/23455'} />
        </BrowserRouter>
      </ThemeProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
