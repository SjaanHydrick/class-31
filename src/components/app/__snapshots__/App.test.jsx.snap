// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <div>
    <div
      class="light"
    >
      <div
        class="header"
      >
        <span
          class="darkMode"
        >
          Dark Mode
        </span>
        <input
          type="checkbox"
        />
        <a
          href="/"
        >
          <h1>
            Popular Television
          </h1>
        </a>
      </div>
    </div>
    <img
      alt="loading"
      class="loading"
      src="https://media2.giphy.com/media/9nmRB7rnndLsk/source.gif"
    />
  </div>
</DocumentFragment>
`;

exports[`App component renders Show 1`] = `
<DocumentFragment>
  <div
    class="light"
  >
    <div
      class="show"
    >
      <img
        class="image"
      />
      <h2
        class="name"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`App component renders ShowDetail 1`] = `
<DocumentFragment>
  <div
    class="light"
  >
    <div
      class="showDetail"
    >
      <img
        class="detailsImg"
      />
      <h2
        class="detailsName"
      />
      <p
        class="status"
      >
        Status: 
      </p>
      <p
        class="description"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`App component renders Shows 1`] = `
<DocumentFragment>
  <div
    class="showList"
    data-testid="shows"
  >
    <div>
      <a
        href="/shows/1"
      >
        <div
          class="light"
        >
          <div
            class="show"
          >
            <img
              alt=""
              class="image"
              src=""
            />
            <h2
              class="name"
            />
          </div>
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
